test/correctness/vectorize_nested.cpp

```diff
@@ -0,0 +1,261 @@
+#include "Halide.h"
+#include "check_call_graphs.h"
+
+using namespace Halide;
+
+int vectorize_2d_round_up() {
+    const int width = 32;
+    const int height = 24;
+
+    Func f("f");
+    Var x("x"), y("y"), xi("xi"), yi("yi");
+
+    f(x, y) = 3 * x + y;
+
+    f.compute_root()
+        .tile(x, y, x, y, xi, yi, 8, 4, TailStrategy::RoundUp)
+        .vectorize(xi)
+        .vectorize(yi);
+
+    Buffer<int> result = f.realize(width, height);
+
+    auto cmp_func = [](int x, int y) {
+        return 3 * x + y;
+    };
+    if (check_image(result, cmp_func)) {
+        return -1;
+    }
+
+    return 0;
+}
+
+int vectorize_2d_guard_with_if() {
+    const int width = 33;
+    const int height = 22;
+
+    Func f("f");
+    Var x("x"), y("y"), xi("xi"), yi("yi");
+
+    f(x, y) = 3 * x + y;
+
+    f.compute_root()
+        .tile(x, y, x, y, xi, yi, 8, 4, TailStrategy::GuardWithIf)
+        .vectorize(xi)
+        .vectorize(yi);
+
+    Buffer<int> result = f.realize(width, height);
+
+    auto cmp_func = [](int x, int y) {
+        return 3 * x + y;
+    };
+    if (check_image(result, cmp_func)) {
+        return -1;
+    }
+
+    return 0;
+}
+
+int vectorize_2d_inlined_with_update() {
+    const int width = 33;
+    const int height = 22;
+
+    Func f, inlined;
+    Var x("x"), y("y"), xi("xi"), yi("yi");
+    RDom r(0, 10, "r");
+    inlined(x) = x;
+    inlined(x) += r;
+    f(x, y) = inlined(x) + 2 * y;
+
+    f.compute_root()
+        .tile(x, y, x, y, xi, yi, 8, 4, TailStrategy::GuardWithIf)
+        .vectorize(xi)
+        .vectorize(yi);
+
+    Buffer<int> result = f.realize(width, height);
+
+    // for (int iy = 0; iy < height; iy++) {
```

Comment:
Instead of adding commented code (4x), I think we should add a `print_image` function (not in this PR) alongside `check_image`, so adding this check is a one liner.

Reply:
Removed debugging code for now, but +1 to print_image function - I'll add it in separate PR.